kythe/java/com/google/devtools/kythe/extractors/java/standalone/Javac9Wrapper.java

```diff
@@ -114,17 +114,35 @@ protected CompilationDescription processCompilation(
 
     String analysisTarget =
         readEnvironmentVariable("KYTHE_ANALYSIS_TARGET", createTargetFromSourceFiles(sources));
-    return javaCompilationUnitExtractor.extract(
-        analysisTarget,
-        sources,
-        classPaths,
-        bootclasspath,
-        sourcePaths,
-        processorPaths,
-        processors,
-        genSrcDir,
-        completeOptions,
-        outputDirectory);
+
+    CompilationDescription desc =
+        javaCompilationUnitExtractor.extract(
+            analysisTarget,
+            sources,
+            classPaths,
+            bootclasspath,
+            sourcePaths,
+            processorPaths,
+            processors,
+            genSrcDir,
+            completeOptions,
+            outputDirectory);
+
+    // Restore output directory, if present. It will be stripped by the extractor but is required
```

Comment:
Why not have the extractor leave the option?

Reply:
It is replaced with a temporary directory to, I suspect, avoid trampling the outputs from the actual compiler but can easily be restored when building the CompilationUnit rather than merely removed.